nio/responses.py

```diff
@@ -265,9 +266,39 @@ def from_dict(cls, parsed_dict):
         except (SchemaError, ValidationError):
             return cls("unknown error")
 
+        if parsed_dict["errcode"] == "M_LIMIT_EXCEEDED":
+            return LimitExceededError.from_dict(parsed_dict)
+
         return cls(parsed_dict["error"], parsed_dict["errcode"])
 
 
+@attr.s
+class LimitExceededError(ErrorResponse):
```

Comment:
I don't think that adding this error class makes much sense given how we use those error classes.

The sole reason why we annoyingly introduce all those classes is that we want to make the life of client users easier. This way they don't need to remember the request type that they made, they know from the response/error type what happened. They mostly handle the same response or error type the same way.

If we introduce this class, a client user will not know what type of request he needs to retry. The better way to handle this is to add the `retry_after_ms` attribute to the base error response class since every type of request can fail this way.

Now granted the AsyncClient doesn't really care about this since he has the luxury of the async/await syntax and the request type is right there on the stack for them.

So you don't need to go the hard way implementing this if you only care about the AsyncClient, you can check the return code of the transport response, verify the response schema and get the retry timeout directly from the transport response content. Wait there for a while and retry the request on behalf of the user. 

We should eventually stop retrying and return an error response but this can be implemented later.